### PR TITLE
careful with small bboxes

### PR DESCRIPTION
**Pull request by esheldon**
some checking to make sure the boxes are OK and non-empty mbobs were created

### CODE REVIEW

File: metadetect/lsst_metadetect.py [hunk 1]
```diff
@@ -128,15 +128,19 @@ def _add_positions_and_psf(self, medsifier, res, shear_str, mbobs_list):
...
+                newres['box_size'][i] = -9999
+
+                newres['row0'][i] = bbox.getBeginY()
+                newres['col0'][i] = bbox.getBeginX()
+                newres['row'][i] = orig_cen.getY() - newres['row0'][i]
+                newres['col'][i] = orig_cen.getX() - newres['col0'][i]
+
+                newres['ormask'][i] = self.ormask[
+                    int(newres['row'][i]), int(newres['col'][i]),
+                ]
```
Comment (beckermr) [inline metadetect/lsst_metadetect.py:143]: Why is this code in the else statement?

**Comment by esheldon** [comment]
@beckermr ready for another look

File: metadetect/lsst_metadetect.py [hunk 2]
```diff
@@ -168,6 +177,10 @@ def _add_positions_and_psf(self, medsifier, res, shear_str, mbobs_list):
             else:
                 newres["mfrac"] = 0
 
+                # mark bad bboxes as masked
+                w, = np.where(newres['flags'] & procflags.BAD_BBOX != 0)
+                newres['mfrac'][w] = 1.0
+
```
Comment (beckermr) [inline metadetect/lsst_metadetect.py:183]: This block confuses me. Is mfrac == 1 even for bad boxes when we have an mfrac image with non-zero values?

File: metadetect/mfrac.py [hunk 3]
```diff
@@ -63,20 +63,23 @@ def measure_mfrac(
     mfracs = []
     for i in range(x.shape[0]):
         try:
-            obs = m.get_obs(i, 0)
-            wgt = obs.weight.copy()
-            msk = (obs.bmask & BMASK_EDGE) != 0
-            wgt[msk] = 0
-            wgt[~msk] = 1
-            obs.set_weight(wgt)
+            if box_sizes[i] > 0:
```
Comment (beckermr) [inline metadetect/mfrac.py:66]: If box_sizes == 0, is that going to break the CatalogMEDSifier?

**Comment by esheldon** [comment]
The idea is to make sure downstream people don't accidentally include these

The flags are also set so it doesn't really matter, people won't use them if they cut flags.

I can remove this

**Comment by esheldon** [comment]
(can't seem to respond to your review comments)

File: metadetect/lsst_metadetect.py (hunk 2 above)
Comment (beckermr) [reply metadetect/lsst_metadetect.py:183]: I think maybe this block should always be run (so move it out of the else block).

**Comment by esheldon** [comment]
doesn't seem to crash the `CatalogMEDSifier`

**Comment by beckermr** [comment]
My questions are really about edge cases with box_size = -9999. Can we add a test for those so we know what happens?

File: metadetect/lsst_metadetect.py (hunk 2 above)
Comment (esheldon) [reply metadetect/lsst_metadetect.py:183]: well, mfrac is set to 1 in the mfrac calculating coded if the bbox is <= 0 so it would be redundant in that case

Comment (esheldon) [reply metadetect/lsst_metadetect.py:183]: I think I'll just remove this and people should use the damn flags

Comment (beckermr) [reply metadetect/lsst_metadetect.py:183]: Ahhhhh. We should not rely on this behavior unless we add a test for it for the mfrac code.

Comment (esheldon) [reply metadetect/lsst_metadetect.py:183]: uhhh.. no because that would be inconsistent with the case that the mfrac code gets run

Comment (beckermr) [reply metadetect/lsst_metadetect.py:183]: Can I push to this branch? I want to avoid a change in one spot depending on behavior in another. If we simply add a test to make sure the mfrac measurements are always 1 if bbox < 0 in the test suite, then the code as is is fine.

**Comment by esheldon** [comment]
I don't actually know how to trigger the bbox problem, it is rare

**Comment by esheldon** [comment]
about 1 in 500 images from the sim that had stars triggered this as I recall

File: metadetect/lsst_metadetect.py (hunk 2 above)
Comment (esheldon) [reply metadetect/lsst_metadetect.py]: go for it

**Review by beckermr — APPROVED** [review]

**Comment by esheldon** [comment]
gonna bump the version and merge this
